backend/server.ts

```diff
@@ -19,53 +21,45 @@ app.use(cors());
-    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
-    region: 'us-east-2',
-});
-
 const storage = multer.memoryStorage(); // Store the file in memory
 const upload = multer({ storage: storage });
 
-app.get('/upload-page', (req, res) => {
+app.get('/upload-page', (req, res) => { // This is just for testing purposes
     res.sendFile(path.join(__dirname, '../Frontend/testwebsite.html'));
 });
 
-app.post('/upload', upload.single('packageContent'), async (req, res, next) => {
+app.post('/package', upload.single('packageContent'), async (req, res) => {
     try {
         if (!req.file) {
-            console.warn('No file provided in the upload.');
+            logger.info("No file provided in the upload.");
             return res.status(400).send('No file uploaded');
         }
 
-        const bucketName = process.env.AWS_S3_BUCKET_NAME;
+        const metadata = await apiPackage.extractMetadataFromZip(req.file.buffer);
+        const url = await apiPackage.getGithubUrlFromZip(req.file.buffer);
+        const githubInfo = apiPackage.parseGitHubUrl(url); 
 
-        if (!bucketName) {
-            console.error('Error: S3 bucket name not configured.');
-            return res.status(500).send('S3 bucket name not configured.');
+        if (!githubInfo) {
+            logger.info("Invalid GitHub repository URL.");
+            return res.status(400).send('Invalid GitHub repository URL.');
         }
-
-        const metadata = await apiPackage.extractMetadataFromZip(req.file.buffer);
+        const { owner, repo } = githubInfo;
+        const action = Action.CREATE;
         await prismaCalls.uploadMetadataToDatabase(metadata);
-
-        const params = {
-            Bucket: bucketName,
-            Key: req.file.originalname, // File name you want to save as in S3
-            Body: req.file.buffer,
-        };
+        await prismaCalls.createPackageHistoryEntry(metadata.ID, 1, action); // User id is 1 for now
+        await apiPackage.calculateAndStoreGithubMetrics(owner, repo);
 
         // Uploading files to the bucket
-        s3.upload(params, function (err: Error, data: S3.ManagedUpload.SendData) {
-            if (err) {
-                console.error('S3 upload error: ', err);
-                return res.status(500).send('Error while uploading to S3.');
-            }
-            res.send(`File uploaded successfully to ${data.Location}`);
-        });
+        const s3Response = await apiPackage.uploadToS3(req.file.originalname, req.file.buffer);
+        res.send(`File uploaded successfully to ${s3Response.Location}`);
     } catch (error) {
-        next(error); // Forward the error to the error handler
+        if (error instanceof Error) {
+            logger.info("S3 upload error: ", error.message);
+            res.status(500).send(error.message);
+        } else {
+            logger.info("Internal Server Error");
+            res.status(500).send("Internal Server Error");
```

Comment:
I think you can just do logger.info("Error in post /package"); and then res.status(500).send("Internal Server Error"). Because in theory the backend functions should just send a more specific error if it's due to a user mistake or something like that.